project1/backend/controllers/queueController.js

```diff
@@ -0,0 +1,42 @@
+const queueModel = require('../models/queueModel.js');
+const catchAsync = require('../utils/catchAsync');
+const AppError = require('../utils/AppError');
+
+exports.enqueuePatient = catchAsync(async (req, res, next) => {
```

Comment:
I think it looks more readable, we use it instead of a try...catch block that adds repeating code and extra indentation. In this case, if we try to promisify it, we need to add a catch to the point where the function is invoked, so it will add unnecessary `f().catch(next)` to the queueRoutes.js file === `router.route('/').post(queueController.enqueuePatient.catch(next));`